src/prof_sys.c

```diff
@@ -502,6 +513,10 @@ prof_dump(tsd_t *tsd, bool propagate_err, const char *filename,
 	buf_writer_terminate(tsd_tsdn(tsd), &buf_writer);
 	prof_dump_close(&arg);
 
+	prof_dump_hook_t dump_hook = prof_dump_hook_get();
+	if (dump_hook) {
```

Comment:
nit: if (dump_hook != NULL)

Question: should we have a similar check for the backtrace hook? Right now if the bt hook is set to NULL, it will crash upon calling. They are a bit different, in a sense that the dump hook is "additional" while the backtrace hook is replacement (so kind of must have).

Reply:
Fixed.

Currently we sample only when we allocate, and we allocate only after malloc is initialized. The initialization calls `prof_hooks_init`, assigning the backtrace hook to the default. Therefore I presume we can omit the check, I will add an assert though.

Reply:
We allow setting the bt hook to null through the mallctl though, in which case an assertion probably makes less sense (i.e. if null bt hook is considered an error, the mallctl should reject such value).

Reply:
Right. What about rejecting null in mallctl? Sampling without backtrace doesn't make sense anyway.

Reply:
That works for me. In that case we can add an assertion to the callsite then.